scripts/install_dev_tools

```diff
@@ -261,3 +265,27 @@ if should_cli_be_installed "promtool" "${cli_tools_arr[@]}" && \
       mv "${VENV}/bin/${DIR_NAME}/promtool" "${VENV}/bin/"
       rmdir "${VENV}/bin/${DIR_NAME}"
 fi
+
+# Conftest CLI
+if should_cli_be_installed "conftest" "${cli_tools_arr[@]}" && \
+    ! [ -x "$(command -v "${VENV}/bin/conftest")" ]; then
+      CONFTEST_CLIENT_URL=$(get_download_url_from_github_api "${CONFTEST_CLIENT_API_URL}") \
+                    || (echo "$CONFTEST_CLIENT_URL"; cleanup_and_exit 2)
+      download_file_from_url "${CONFTEST_CLIENT_URL}" "${DWN_DIR}"
+      CONFTEST_CLIENT_PATH="${DWN_DIR}"/$(basename "${CONFTEST_CLIENT_URL}")
+      extract_file_to_dir "${CONFTEST_CLIENT_PATH}" "${VENV}/bin/" "conftest"
+fi
+
+# Install bats via repository
+if should_cli_be_installed "bats" "${cli_tools_arr[@]}" && \
+    ! [ -x "$(command -v "${VENV}/bin/bats")" ]; then
+    if ! [ -x "$(command -v git)" ]; then
+        echo "git command not found"
+        exit 2
+    fi
+    git clone "${BATS_REPO_URL}" "${DWN_DIR}/bats"
+    pushd "${DWN_DIR}/bats"
+    ./install.sh "../../"
```

Comment:
This will be relative to the `DWN_DIR` from `mktemp`, which is system-dependent. This didn't work on macOS.

Should we pass the venv as the target?

The other thing is-- if this is a directory under `/tmp`, wouldn't this install it to the root fs? Shouldn't that fail unless this script is run as root?

Reply:
Where was the directory created ?

The idea is to create directory inside .venv/ and not the /tmp/ target, that is why we have before calling mkdir:

TMPDIR="${VENV}"

https://github.com/konveyor/pelorus/blob/master/scripts/install_dev_tools#L206

This should work on Mac OS X as well. Unless you have other setting specified such as _CS_DARWIN_USER_TEMP_DIR or _CS_DARWIN_USER_TEMP_DIR

But I will change it to pass venv, will be more robust

Reply:
Secondly this script should not be ran as root, we really don't want that.

Reply:
@KevinMGranger ok in the newest revision I have changed two things:

To ensure download dir is created always in the .venv regardless of the system env I am passing this directly to the mktemp(1) command rather then relying on the env variable:

```
> diff
-DWN_DIR=$(TMPDIR="${VENV}" mktemp -d -t "${TMP_DIR_PREFIX}XXXXX") || exit 2
+DWN_DIR=$(mktemp --tmpdir="${VENV}" -d -t "${TMP_DIR_PREFIX}XXXXX") || exit 2

```

Secondly to calculate absolute path of the place where bats is installed I am ensuring script enters this directory first and then takes this value from simple pwd(1). This should be platform independent/works anywhere.